harness/src/search/search.ts

```diff
@@ -0,0 +1,233 @@
+import fs from 'fs';
+import path from 'path';
+import dotenv from "dotenv";
+import { ethers, JsonRpcProvider } from "ethers";
+import { getFullBlock, getRawReceipt, getNumBytes, objectToRlp, bytes32 } from "@axiom-crypto/tools";
+import { AddressId, RcId, StorageId, TxId } from "../types";
+import { BLOCK_INTERVAL, BLOCK_SAMPLES, data, matchIgnoreAddrs } from "./defaults";
+dotenv.config();
+
+// // Block numbers to include in the search (useful if certain events happened at specific blocks)
+// const INCLUDE_BLOCKS: number[] = [
+//   8285121,
+// ];
+
+export const search = async (
+  options: {
+    provider: string;
+    samples?: number;
+    interval?: number;
+    block?: number;
+    include?: string;
+    ignore?: string;
+    output?: string;
+  }
+) => {
+  const provider = new ethers.JsonRpcProvider(options.provider);
+  let currentBlock = await provider.getBlockNumber();
+  const numSamples = options.samples ?? BLOCK_SAMPLES;
+  const interval = options.interval ?? BLOCK_INTERVAL;
+
+  let includeBlocks: number[] = [];
+  if (options.include !== undefined) {
+    includeBlocks = options.include.split(",").map(Number);
+  }
+  let outputPath = '../output';
+  if (options.output !== undefined) {
+    outputPath = options.output;
+  }
+
+  let ignoreAddrs: string[] = [...matchIgnoreAddrs];
+
+  const chainId = (await provider.getNetwork()).chainId.toString();
+  data.chainId = chainId;
+  if (options.block !== undefined){ 
+    currentBlock = options.block;
+  }
+  data.blockRange.start = currentBlock;
+  data.blockRange.end = currentBlock;
+
+  let blocksToProcess: number[] = [...includeBlocks];
+  for (let i = 0; i < numSamples; i++) {
+    blocksToProcess.push(currentBlock - i * interval);
+  }
+  for (let i = 0; i < blocksToProcess.length; i++) {
+    console.log("Currently processing block:", blocksToProcess[i]);
+    const blockNumber = blocksToProcess[i];
+    if (blockNumber < data.blockRange.start) {
+      data.blockRange.start = blockNumber;
+    }
+    
+    const block = await getFullBlock(provider, blockNumber);
+    // console.log(blockNumber, block)
+
+    if (!block || block.transactions.length === 0) {
+      continue;
+    }
+    // Pick random transaction index from block of account to include
+    const idx = Math.floor(Math.random() * block.transactions.length);
```

Comment:
Does the random choosing give good enough results when searching?

Reply:
yeah since this is just for accounts it's fine